src/app/components/Filters/FieldsetDate.jsx

```diff
@@ -1,27 +1,100 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import Actions from '../../actions/Actions.js';
+import {
+  extend as _extend,
+} from 'underscore';
+
+class FieldsetDate extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dateAfter: '0',
+      dateBefore: '0',
+      selectedFacets: this.props.selectedFacets,
+    };
+
+    this.inputChange = this.inputChange.bind(this);
+  }
+
+  /**
+   * inputChange()
+   * Updates the states and the store based on the input values.
+   *
+   */
+  inputChange(e) {
+    const value = (e.target.value).replace(/[a-zA-Z]/g, '');
+    let displayValue = '';
+
+    if (e.target.name === 'start-date') {
+      displayValue = 'dateAfter';
+    } else if (e.target.name === 'end-date') {
+      displayValue = 'dateBefore';
+    }
+
+    if (displayValue) {
+      this.setState(
+        {
+          [displayValue]: value,
+          selectedFacets: _extend(
+            this.state.selectedFacets,
+            { [displayValue]: value }
+          ),
+        },
+        () => {
+          Actions.updateSelectedFacets(this.state.selectedFacets);
```

Comment:
This is okay for now, but what you can do is bubble up the value to the parent `FilterPopup` component and pass the date value. Then the form can handle all the data and store values and this component can just render/pass data along. If you do this, then you don't have to bring in the `selectedFacets` as a prop and this component can only worry about its date value (which I think would be more appropriate).

Reply:
Can you also temporarily remove this? It breaks the app I think because the date object is not completely set up to filter yet.

Reply:
sure.